### ISSUE TITLE
Issue with Filter by Facebook Sync Setting

### ISSUE THREAD

**Issue by sukafia** [woocommerce/facebook-for-woocommerce#2616]
## ✍️ Describe the bug

Facebook for WooCommerce ads the "Filter by Facebook sync setting" option for filtering products by Facebook sync setting in the All Products screen. However, this filter isn't correctly filtering all products associated with the sync settings.

The filter does not work on all products. For instance, when some products are assigned the  “Do not sync” setting, they do not appear when filtered with that option.

- [This user ](https://wordpress.org/support/topic/facebook-for-woocommerce-not-syncing-and-filter-is-not-working/)recorded a screencast of the issue [here](https://drive.google.com/file/d/1R8RAd2-qA7SlwaB8_YSND7Qq-dJ4fgkY/view?usp=sharing).

I tested this on my end and it seems the product status may play a role (e.g. draft, private, etc.)

### Affected users

https://wordpress.org/support/topic/filter-by-facebook-sync-setting-not-working-in-admin-all-products
https://wordpress.org/support/topic/facebook-for-woocommerce-not-syncing-and-filter-is-not-working/

## 🚶‍♀️ Steps to reproduce

1. Assigned different sync settings to products (e.g. Sync and show, sync and hide, do not sync)
2. Filter the products by the assigned sync status
3. Notice some products matching the sync setting don't show when filtered by that sync setting (I even noticed a product showing up for an unrelated sync setting)

## ✔️ Expected behavior

Products associated with a sync setting should show up when filtered by that sync setting.

**Comment by sukafia** [comment]
I noticed something interesting while testing this

![wgYTY3.png](https://github.com/woocommerce/facebook-for-woocommerce/assets/60121732/1660caa4-4125-4c17-b060-696477569d31)

Product sync settings "Sync and hide":

![kLjuln.png](https://github.com/woocommerce/facebook-for-woocommerce/assets/60121732/fce3cce6-1052-4c2b-bb73-6d0b44928666)

The Facebook sync status shown in the product interface appears to be due to being out of stock/private. While the assigned status is "Sync and hide".  It's correct for the product to appear when filtered with "Sync and hide" but this initially confused me to think a product with a "Do not sync" status was being shown for the sync and hide filter. 

From this, I gathered part of the issue is the default sync status assigned due to the product status (draft, out of stock, private, etc.) and the assigned sync setting on the product editing screen.
